services/storage/tests/test_dsm_soft_links.py

```diff
@@ -0,0 +1,133 @@
+# pylint: disable=redefined-outer-name
+# pylint: disable=unused-argument
+# pylint: disable=unused-variable
+
+from typing import Iterable
+
+import attr
+import pytest
+from aiopg.sa.engine import Engine
+from simcore_service_storage.dsm import DataStorageManager
+from simcore_service_storage.models import FileMetaData, FileMetaDataEx, file_meta_data
+from simcore_service_storage.settings import SIMCORE_S3_STR
+from simcore_service_storage.utils import create_resource_uuid
+from sqlalchemy.sql.expression import literal_column
+
+
+@pytest.fixture
+def storage(postgres_engine: Engine) -> DataStorageManager:
+
+    return DataStorageManager(
+        s3_client=None,
+        engine=postgres_engine,
+        loop=None,
+        pool=None,
+        simcore_bucket_name="master-simcore",
+        has_project_db=True,
+        app=None,
+    )  # type: ignore
+
+
+@pytest.fixture()
+async def output_file(
+    user_id: int, project_id: str, postgres_engine: Engine
+) -> Iterable[FileMetaData]:
+
+    node_id = "fd6f9737-1988-341b-b4ac-0614b646fa82"
+
+    # pylint: disable=no-value-for-parameter
+
+    file = FileMetaData()
+    file.simcore_from_uuid(
+        f"{project_id}/{node_id}/filename.txt", bucket_name="master-simcore"
+    )
+    file.entity_tag = "df9d868b94e53d18009066ca5cd90e9f"
+    file.user_name = "test"
+    file.user_id = str(user_id)
+
+    async with postgres_engine.acquire() as conn:
+        stmt = (
+            file_meta_data.insert()
+            .values(
+                **attr.asdict(file),
+            )
+            .returning(literal_column("*"))
+        )
+        result = await conn.execute(stmt)
+        row = await result.fetchone()
+
+        # hacks defect
+        file.user_id = str(user_id)
+        file.location_id = str(file.location_id)
```

Comment:
could you clarify the comment please?

Reply:
haha! There is not solid justification for this hack except that the entire DSM object is a mash-potato that will slowly refactor :-)  
This overcomes an inconsistency so I can get the comparisons in the test right. It is not relevant here since it is a test. But it should disappear as we refactor DataStorageManager class